### ISSUE TITLE
Handling negative values in the `zoom` section for `LINK_GOTO` in `linkDest`

### ISSUE THREAD

**Issue by ckwastra** [pymupdf/PyMuPDF#2341]
## Describe the bug

I encountered an issue with the `linkDest` class when dealing with a PDF that has internal links with negative values in the zoom section. One such link has the following properties:

```python
{
    'from': Rect(72.55999755859375, 209.0, 414.55999755859375, 224.0),
    'id': '',
    'kind': 4,
    'name': 'page=12&zoom=0,43,-68',
    'xref': 1051
}

```
This link should be a `LINK_GOTO` link, but due to the regular expression used to parse the URI in the `linkDest` class, it was incorrectly identified as a `LINK_NAMED` link. The issue lies in the following line:

```python
m = re.match('^#page=([0-9]+)&zoom=([0-9.]+),([0-9.]+),([0-9.]+)$', self.uri)
```

As a workaround, I suggest updating the regular expression to allow for optional negative signs in front of the x and y coordinates. The modified regular expression would look like this:

```python
m = re.match('^#page=([0-9]+)&zoom=([0-9.]+),(-?[0-9.]+),(-?[0-9.]+)$', self.uri)
```

## To Reproduce

To reproduce this issue, follow these steps:

1. Create a PDF with an internal link that has a negative value in the `zoom` section.
2. Use the following minimal code snippet to extract the link information using PyMuPDF:
   ```python
   import fitz

   filename = "path/to/your/pdf"
   doc = fitz.open(filename)
   page = doc[0]
   links = page.get_links()

   for link in links:
       print(link)
   ```
3. Observe that the link is incorrectly identified as a `LINK_NAMED` link, when it should be a `LINK_GOTO` link.

## Your configuration

- Operating System: macOS Catalina 10.15.7 64-bit
- Python version: Python 3.11.2 64-bit
- PyMuPDF version: PyMuPDF 1.21.1, installed via pip (wheel)

```shell
>>> print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)
3.11.2 (main, Mar  2 2023, 06:01:29) [Clang 12.0.0 (clang-1200.0.32.29)] 
 darwin 
 
PyMuPDF 1.21.1: Python bindings for the MuPDF 1.21.1 library.
Version date: 2022-12-13 00:00:01.
Built for Python 3.11 on darwin (64-bit).
```

## Additional context

I also looked into the PDF standard, which states that the presence of negative values in the `zoom` section should be legal. I was able to open this link correctly in Adobe Acrobat.

>**12.3.2.2 Explicit Destinations**
Table 151 shows the allowed syntactic forms for specifying a destination explicitly in a PDF file. In each case, _page_ is an indirect reference to a page object (except in a remote go-to action; see 12.6.4.3, “Remote Go-To Actions”). All coordinate values (_left_, _right_, _top_, and _bottom_) shall be expressed in the default user space coordinate system. The page’s _bounding box_ is the smallest rectangle enclosing all of its contents. (If any side of the bounding box lies outside the page’s crop box, the corresponding side of the crop box shall be used instead; see 14.11.2, “Page Boundaries,” for further discussion of the crop box.)

**Comment by JorjMcKie** [comment]
@ckwastra - your last comment mentions features that do not seem to be supported by the PDF language spec including version 1.7.

**Comment by JorjMcKie** [comment]
> Also I believe that syntax `/FitR left bottom right top` in the PDF spec corresponds to the `viewrect=left,top,wd,ht` parameter in the reference I provided earlier.

Ah, I knew that one of course. I was referring to the "viewrect" keywords.

**Comment by julian-smith-artifex-com** [comment]
We've just pushed a commit with the updated regex's as you suggested, thanks for including them in this issue.

Do you have a PDF that you can share, that has an internal link that has a negative value in the zoom section? Would allow us to make a test.

**Comment by julian-smith-artifex-com** [comment]
Thanks for this, but that PDF's copyright probably isn't ok for us to include it in PyMuPDF's git, or on this issues page.

I think the change to the regex is simple enough that it'll be ok to not have an automated test for it anyway.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.5.
